pkg/core/policy/dataplane_matcher.go

```diff
@@ -49,6 +49,56 @@ func SelectDataplanePolicy(dataplane *mesh.DataplaneResource, policies []Datapla
 	return bestPolicy
 }
 
+// SelectInboundDataplanePolicies given a Dataplane definition and a list of DataplanePolicy returns the "best matching" DataplanePolicy for each inbound separately.
+// A DataplanePolicy for an inbound is considered a match if the inbound matches all the tags listed in selector of the DataplanePolicy.
```

Comment:
Should tags have weights?
I mean should `kuma.io/zone` should be outweighting `app/version` ?

Reply:
We could, I think we even talked about this once, but
1) That complicates the matching mechanism, we would need to add some parameterization of tag weights - seems like a breaking change
2) Definitely out of scope of this PR, since it touches all matchers

If I were to improve something about the matching mechanism in Kuma, I'd try to support negate cases or aggregation of many policies instead of "only match the most specific"